src/SourceWatcherFuncApp/Utilities/EntityHelper.cs

```diff
@@ -27,16 +29,90 @@ public static DetectorEntity PrepareEntityForLoad(Stream streamAssemblyData, str
                 invoker.InitializeEntryPoint(temp);
                 var resourceFilter = invoker.ResourceFilter;
                 detectorPackage.IsInternal = resourceFilter != null? resourceFilter.InternalOnly : false;
-                detectorPackage.ResourceType = resourceFilter != null ? resourceFilter.ResourceType.ToString() : "";
                 detectorPackage.SupportTopicList = invoker.EntryPointDefinitionAttribute != null ? invoker.EntryPointDefinitionAttribute.SupportTopicList : new List<SupportTopic>() ;
                 detectorPackage.AnalysisTypes = invoker.EntryPointDefinitionAttribute != null ? invoker.EntryPointDefinitionAttribute.AnalysisTypes : new List<string>();
+
+                if(invoker.ResourceFilter != null)
+                {
+                    if (invoker.ResourceFilter is AppFilter)
+                    {
+                        // Store WebApp related info
+                        var resourceInfo = invoker.ResourceFilter as AppFilter;
+
+                        AppType appType = resourceInfo.AppType;
+                        var appTypesList = Enum.GetValues(typeof(AppType)).Cast<AppType>().Where(p => appType.HasFlag(p)).Select(x => Enum.GetName(typeof(AppType), x));
+                        detectorPackage.AppType = string.Join(",", appTypesList);
+
+                        PlatformType platformType = resourceInfo.PlatformType;
+                        var platformTypesList = Enum.GetValues(typeof(PlatformType)).Cast<PlatformType>().Where(p => platformType.HasFlag(p)).Select(x => Enum.GetName(typeof(PlatformType), x));
+                        detectorPackage.PlatForm = string.Join(",", platformTypesList);
+
+                        StackType stackType = resourceInfo.StackType;
+                        var stackTypesList = Enum.GetValues(typeof(StackType)).Cast<StackType>().Where(s => stackType.HasFlag(s)).Select(x => Enum.GetName(typeof(StackType), x));
+                        detectorPackage.StackType = string.Join(",", stackTypesList);
+
+                        detectorPackage.ResourceProvider = "Microsoft.Web";
+                        detectorPackage.ResourceType = "sites";
+                    } 
+                    else if (invoker.ResourceFilter is ApiManagementServiceFilter)
```

Comment:
Since this function app is throw-away code you don't have to do this suggestion but if you like one thing you can do to avoid this huge if else body and save time in maintaining this code is reference the Diagnostics.ModelAndUtils project and use my example code below. This code is simply reusing the `IsApplicable` method in the IResource classes to determine if that `IResourceFilter` is applicable. This is similar to what we do today for list detectors call. https://github.com/Azure/Azure-AppServices-Diagnostics/blob/7b4bd51ad8b0c228e2c7a85421e755714c328556/src/Diagnostics.RuntimeHost/Services/CacheService/InvokerCacheService.cs#L49-L59

                var services = Assembly.GetExecutingAssembly().GetTypes().Where(t => t.GetInterfaces().Contains(typeof(IResource)) && t.IsAssignableFrom(typeof(Attribute)));
                foreach (var item in services)
                {
                    IResource resource = (IResource)Activator.CreateInstance(item);
                    if (resource.IsApplicable(invoker.ResourceFilter))
                    {
                        detectorPackage.ResourceProvider = resource.Provider;
                        detectorPackage.ResourceType = resource.ResourceType.ToString();
                    }
                }